src/librustdoc/clean/types.rs

```diff
@@ -659,7 +659,10 @@ impl Item {
                     asyncness: hir::IsAsync::NotAsync,
                 }
             }
-            ItemKind::FunctionItem(_) | ItemKind::MethodItem(_, _) | ItemKind::TyMethodItem(_) => {
+            ItemKind::FunctionItem(_)
+            | ItemKind::MethodItem(_, _)
+            | ItemKind::TyMethodItem(_)
+            | ItemKind::TestItem(_) => {
```

Comment:
Modelling a `#[test] fn` as a distinct `ItemKind` seems a bit strange: `#[test] fn`s are still function items, I wonder if you can distinguish non-`#[test]` fns versus `#[test]` fns by looking at the existence/absence of the `#[test]` attribute.

But yeah, rendering docs for tests do seem useful. Though a question that immediately comes to my mind is that do test-affiliated module docs get rendered too? As in e.g. docs on `#[cfg(test)] mod tests;`. I would think that they do, but I haven't looked closely if they do in this implementation.

Reply:
Yes, they are rendered in this implementation though they are most likely not `pub` therefore the additional flag `--document-private-items` may be needed.

Reply:
I agree with @jieyouxu, but also I like this difference as it makes it simpler to know what's being manipulated...

As for `cfg`ed out items, unfortunately you will need to pass `--cfg test` to have them (alongside `--document-private-items`).